### ISSUE TITLE
Use KEFS version of boot for neutron.img ?

### ISSUE THREAD

**Issue by bibikalka1** [neutronscott/flip2#28]
@neutronscott 

I noticed that the older KEEZ version of boot is used in neutron.img

Would it make sense to update that to KEFS boot? What are the mods you have in neutron ? Is it just the magisk'ed boot image, or is there more to it?

Anyway, my flip2 has KEFS, so I flashed your custom recovery, and dumped stock boot from it. Please see the attached.
[boot_kefs.zip](https://github.com/user-attachments/files/17013799/boot_kefs.zip)

**Comment by neutronscott** [comment]
I made one and was told it didn't work. I tried to flash on a Us cellular device and wrote a bad preloader I think so I can't test anymore.

**Comment by neutronscott** [comment]
Oh and there's one extra piece. It was Magisk applied and a custom init rc that had 

```
on post-fs-data
  exec u:r:magisk:s0 root root -- ${MAGISKTMP}/magisk resetprop -n ro.vendor.tct.endurance true
```

**Comment by bibikalka1** [comment]
> I made one and was told it didn't work. I tried to flash on a Us cellular device and wrote a bad preloader I think so I can't test anymore.

Strange - it's quite benign if only boot is written to, I will test just the modded boot then! If you have your modded version for TracFone - do share please.

**Comment by bibikalka1** [comment]
@neutronscott 

On the 2nd thought, I will simply flash your neutron boot, then will run magisk on kefs kernel, and will use the kitchen to add the script on PC to this kefs boot.

Then will flash this new kernel, and will see how it goes!

**Comment by neutronscott** [comment]
I made a kefs branch using your supplied .zip here: https://github.com/neutronscott/flip2/releases/tag/v2024.09.17.kefs

**Comment by bibikalka1** [comment]
> I made a kefs branch using your supplied .zip here:

Appreciate!!! This way it's even easier for me :)

Will test and report back, probably on Friday

**Comment by bibikalka1** [comment]
@neutronscott 

Alright, tested  your kefs version of neutron.img, all works!

Followed this guide to enable adb, etc:
https://www.reddit.com/r/dumbphones/comments/1cl82lb/tcl_flip_2_t408dl_usb_debugging_root_and_apk/

Magisk version 27.0 works fine, Zgysk works too. Got GAPPS working:
https://www.reddit.com/r/dumbphones/comments/1fg2ihp/tcl_flip_2_debug_enable_guide_gapps_guide_if_you/

Used ATT launcher from here:
https://drive.google.com/file/d/1GvitTaytEQks2CJxfSEs8k_F1bUyBh-r/view

For your collection, here is the kefs stock recovery - although it's not really needed, since your prior modded recovery can be used without issues:
[recovery_kefs.zip](https://github.com/user-attachments/files/17063571/recovery_kefs.zip)

**Comment by bibikalka1** [comment]
**Ran gdisk and mmc info on Flip 2:**
[gdisk01.txt](https://github.com/user-attachments/files/17069169/gdisk01.txt)
[mmc32.txt](https://github.com/user-attachments/files/17069170/mmc32.txt)

**Comment by flipphone2** [comment]
> @neutronscott
> 
> I noticed that the older KEEZ version of boot is used in neutron.img
> 
> Would it make sense to update that to KEFS boot? What are the mods you have in neutron ? Is it just the magisk'ed boot image, or is there more to it?
> 
> Anyway, my flip2 has KEFS, so I flashed your custom recovery, and dumped stock boot from it. Please see the attached. [boot_kefs.zip](https://github.com/user-attachments/files/17013799/boot_kefs.zip)

Are you talking about the recovery2.img?
Can you pull the other images? Such as the super image, vbmeta, vbmeta_system, vbmeta_vendor.
Thanks.

**Comment by flipphone2** [comment]
> **Ran gdisk and mmc info on Flip 2:** [gdisk01.txt](https://github.com/user-attachments/files/17069169/gdisk01.txt) [mmc32.txt](https://github.com/user-attachments/files/17069170/mmc32.txt)

Is this from KEFS? Or are they all the same regardless?

**Comment by bibikalka1** [comment]
> > **Ran gdisk and mmc info on Flip 2:** [gdisk01.txt](https://github.com/user-attachments/files/17069169/gdisk01.txt) [mmc32.txt](https://github.com/user-attachments/files/17069170/mmc32.txt)
> 
> Is this from KEFS? Or are they all the same regardless?

@flipphone2 
Yeah, these are the same. I just thought it'd be useful to someone if I dumped them.

For KEFS full dump - the biggest issue is where to store it on the web! What do you need this for? It's easy to dump it, I could probably closer to the end of the week if you need it.

**Comment by chungenhung** [comment]
> > > **Ran gdisk and mmc info on Flip 2:** [gdisk01.txt](https://github.com/user-attachments/files/17069169/gdisk01.txt) [mmc32.txt](https://github.com/user-attachments/files/17069170/mmc32.txt)
> > 
> > 
> > Is this from KEFS? Or are they all the same regardless?
> 
> @flipphone2 Yeah, these are the same. I just thought it'd be useful to someone if I dumped them.
> 
> For KEFS full dump - the biggest issue is where to store it on the web! What do you need this for? It's easy to dump it, I could probably closer to the end of the week if you need it.

I found the super image for KEFS.
Now I would like to have the stock boot image, vbmeta, vbmeta_system, vbmeta_vendor.
Can you use google drive?
I am having a weird issue with some phones after flashing them. Basically I think they can't make calls on VoLTE, but the connection show 4G LTE when not in a call, and web browsing is OK too.
I've flashed them with KEFH stock room, and still having the issue.
Something must have changed in other area of the phone (other than the boot, vbmeta, super), but I am new to this and not able to figure out what got changed.

**Comment by flipphone2** [comment]
> > > **Ran gdisk and mmc info on Flip 2:** [gdisk01.txt](https://github.com/user-attachments/files/17069169/gdisk01.txt) [mmc32.txt](https://github.com/user-attachments/files/17069170/mmc32.txt)
> > 
> > 
> > Is this from KEFS? Or are they all the same regardless?
> 
> @flipphone2 Yeah, these are the same. I just thought it'd be useful to someone if I dumped them.
> 
> For KEFS full dump - the biggest issue is where to store it on the web! What do you need this for? It's easy to dump it, I could probably closer to the end of the week if you need it.

I found the stock KEFS super image.
Now I would like to have the stock boot image, vbmeta, vbmeta_system, vbmeta_vendor.
Can you use google drive?
I am having a weird issue with some phones after flashing them. Basically I think they can't make calls on VoLTE, but the connection show 4G LTE when not in a call, and web browsing is OK too.
I've flashed them with KEFH stock room, and still having the issue.
Something must have changed in other area of the phone (other than the boot, vbmeta, super), but I am new to this and not able to figure out what got changed.

**Comment by C7CodingDev** [comment]
> > > > **Ran gdisk and mmc info on Flip 2:** [gdisk01.txt](https://github.com/user-attachments/files/17069169/gdisk01.txt) [mmc32.txt](https://github.com/user-attachments/files/17069170/mmc32.txt)
> > > 
> > > 
> > > Is this from KEFS? Or are they all the same regardless?
> > 
> > 
> > [@flipphone2](https://github.com/flipphone2) Yeah, these are the same. I just thought it'd be useful to someone if I dumped them.
> > For KEFS full dump - the biggest issue is where to store it on the web! What do you need this for? It's easy to dump it, I could probably closer to the end of the week if you need it.
> 
> I found the stock KEFS super image. Now I would like to have the stock boot image, vbmeta, vbmeta_system, vbmeta_vendor. Can you use google drive? I am having a weird issue with some phones after flashing them. Basically I think they can't make calls on VoLTE, but the connection show 4G LTE when not in a call, and web browsing is OK too. I've flashed them with KEFH stock room, and still having the issue. Something must have changed in other area of the phone (other than the boot, vbmeta, super), but I am new to this and not able to figure out what got changed.

I have been looking at this thread for a while and have been wondering some things:
I have a KEFS tcl flip2. Should I use the new image or the old one for now? Thanks!

**Comment by bibikalka1** [comment]
> I have been looking at this thread for a while and have been wondering some things:
> I have a KEFS tcl flip2. Should I use the new image or the old one for now? Thanks!

Obviously yes, that's the most consistent. It may not be too bad to run a different version, but why?

**Comment by cooldudeseven7Dev** [comment]
> > I have been looking at this thread for a while and have been wondering some things:
> > I have a KEFS tcl flip2. Should I use the new image or the old one for now? Thanks!
> 
> Obviously yes, that's the most consistent. It may not be too bad to run a different version, but why?

_**I am the Alternate account of C7CodingDev**_
Hi! I ran the KEFS version, but still cant enable ADB, as i mentioned in my new issue: #39 . Do you know any workarounds? Thanks!
